File: PetFinderApp/PetFinderApp/Models/Notification.swift
//
//  Notification.swift
//  PetFinderApp
//
//  Created by Вилина Ольховская on 07.09.2025.
//

import Foundation

struct AppNotification: Decodable {
    let id: String
    let type: String
    let petId: String?
    let sightingId: String?
    let title: String
    let body: String
    let isRead: Bool
    let createdAt: Date
}

File: backend/internal/model/notification.go
package model

import "time"

type Notification struct {
	ID        string    `json:"id"`
	UserID    string    `json:"user_id"`
	Type      string    `json:"type"`
	PetID      *string   `json:"pet_id"`
	SightingID *string   `json:"sighting_id"`
	Title      string    `json:"title"`
	Body      string    `json:"body"`
	IsRead    bool      `json:"is_read"`
	CreatedAt time.Time `json:"created_at"`
}
